src/wp-includes/blocks.php

```diff
@@ -1286,16 +1294,34 @@ function insert_hooked_blocks_into_rest_response( $response, $post ) {
 			'ignoredHookedBlocks' => $ignored_hooked_blocks,
 		);
 	}
-	$content = get_comment_delimited_block_content(
-		'core/navigation',
-		$attributes,
-		$response->data['content']['raw']
+
+	$post_type_to_wrapper_block_mappings = array(
+		'wp_navigation' => 'core/navigation',
+		'wp_post'       => 'core/post-content',
 	);
 
-	$content = apply_block_hooks_to_content( $content, $post );
+	if ( isset( $post_type_to_wrapper_block_mappings[ $post->post_type ] ) ) {
+		$wrapper_block_type = $post_type_to_wrapper_block_mappings[ $post->post_type ];
 
-	// Remove mock Navigation block wrapper.
-	$content = remove_serialized_parent_block( $content );
+		$content = get_comment_delimited_block_content(
+			$wrapper_block_type,
+			$attributes,
+			$response->data['content']['raw']
+		);
+	} else {
+		$content = $response->data['content']['raw'];
+	}
+
+	$content = apply_block_hooks_to_content(
+		$content,
+		$post,
+		'insert_hooked_blocks_and_set_ignored_hooked_blocks_metadata'
+	);
```

Comment:
We're now inserting hooked blocks _and_ setting `ignoredHookedBlocks` metadata. Previously, we were only inserting hooked blocks (i.e. the default behavior of `apply_block_hooks_to_content` when no third argument is given).

This change also affects the Navigation block. We need to make sure that any hooked blocks inside of the Navigation block -- both as first/last child of `core/navigation` itself, but also `after`/`before` blocks _inside_ of the Navigation block -- still work (both on the frontend and in the editor.

Reply:
To test if the Navigation block still works, we can follow the testing instructions from https://github.com/WordPress/gutenberg/pull/59021.

Reply:
Turns out there was an unrelated issue with this. I've filed a fix: https://github.com/WordPress/wordpress-develop/pull/7941

Reply:
I've committed #7941 to Core [`[62639]`](https://core.trac.wordpress.org/ticket/62639), and rebased this PR to include the fix.